solutions/agents/variables.tf

```diff
@@ -0,0 +1,222 @@
+  default     = "private"
+  validation {
+    error_message = "The specified endpoint_type can be private or public only."
+    condition     = contains(["private", "public"], var.log_analysis_endpoint_type)
+  }
+}
+
+variable "log_analysis_agent_custom_line_inclusion" {
+  description = "Log Analysis agent custom configuration for line inclusion setting LOGDNA_K8S_METADATA_LINE_INCLUSION. See https://github.com/logdna/logdna-agent-v2/blob/master/docs/KUBERNETES.md#configuration-for-kubernetes-metadata-filtering for more info."
+  type        = string
+  default     = "label.app.kubernetes.io/name:sample-app\\, annotation.user:sample-user"
+}
+
+variable "log_analysis_agent_custom_line_exclusion" {
+  description = "Log Analysis agent custom configuration for line exclusion setting LOGDNA_K8S_METADATA_LINE_EXCLUSION. See https://github.com/logdna/logdna-agent-v2/blob/master/docs/KUBERNETES.md#configuration-for-kubernetes-metadata-filtering for more info."
+  type        = string
+  default     = "namespace:default"
+}
+
+variable "log_analysis_agent_name" {
+  description = "Log Analysis agent name. Used for naming all kubernetes and helm resources on the cluster."
+  type        = string
+  default     = "logdna-agent"
+  nullable    = false
+}
+
+variable "log_analysis_agent_namespace" {
+  type        = string
+  description = "Namespace where to deploy the Log Analysis agent. Default value is 'ibm-observe'."
+  default     = "ibm-observe"
+  nullable    = false
+}
+
+variable "log_analysis_agent_tolerations" {
+  description = "List of tolerations to apply to Log Analysis agent."
+  type = list(object({
+    key               = optional(string)
+    operator          = optional(string)
+    value             = optional(string)
+    effect            = optional(string)
+    tolerationSeconds = optional(number)
+  }))
+  default = [{
+    operator = "Exists"
+  }]
+}
+
+##############################################################################
+# Cloud Monitoring variables
+##############################################################################
+
+variable "cloud_monitoring_enabled" {
+  type        = bool
+  description = "Deploy IBM Cloud Monitoring agent if set as true."
+  default     = true
+}
+
+variable "cloud_monitoring_access_key" {
+  type        = string
+  description = "Access key used by the IBM Cloud Monitoring agent to communicate with the instance."
+  sensitive   = true
+  default     = null
+}
+
+variable "cloud_monitoring_secret_name" {
+  type        = string
+  description = "The name of the secret which will store the access key."
+  default     = "sysdig-agent"
+  nullable    = false
+}
+
+variable "cloud_monitoring_instance_region" {
+  type        = string
+  description = "The IBM Cloud Monitoring instance region. Used to construct the ingestion endpoint."
+  default     = "us-south"
```

Comment:
same concern here as above comment for logging region. Whatever approach we go with, lets just be consistent